bumble/drivers/intel.py

```diff
@@ -0,0 +1,343 @@
+# -----------------------------------------------------------------------------
+# HCI Commands
+# -----------------------------------------------------------------------------
+HCI_INTEL_READ_VERSION_COMMAND = hci_vendor_command_op_code(0xFC05)  # type: ignore
+HCI_INTEL_SECURE_SEND_COMMAND = hci_vendor_command_op_code(0xFC09)  # type: ignore
+HCI_INTEL_RESET_COMMAND = hci_vendor_command_op_code(0xFC01)  # type: ignore
+HCI_Command.register_commands(globals())
+
+
+@HCI_Command.command(  # type: ignore
+    fields=[("param", 1)],
+    return_parameters_fields=[
+        ("status", STATUS_SPEC),
+        ("data", "*"),
+    ],
+)
+class HCI_Intel_Read_Version_Command(HCI_Command):
+    pass
+
+
+@HCI_Command.command(  # type: ignore
+    # fields=[("param", MAX_FRAGMENT_PAYLOAD + 1)],
+    fields=[("param", "*")],
+    return_parameters_fields=[
+        ("data", "*"),
+    ],
+)
+class Hci_Intel_Secure_Send_Command(HCI_Command):
+    pass
+
+
+@HCI_Command.command(  # type: ignore
+    fields=[
+        ("reset_type", 1),
+        ("patch_enable", 1),
+        ("ddc_reload", 1),
+        ("boot_option", 1),
+        ("boot_param", 4),
+    ],
+    return_parameters_fields=[
+        ("data", "*"),
+    ],
+)
+class Hci_Intel_Reset_Command(HCI_Command):
+    pass
+
+
+# -----------------------------------------------------------------------------
+
+
+@dataclass
+class IntelTLV:
+    type: int
+    lenght: int
+    value: bytearray
+
+
+@dataclass
+class IntelVersionTlv:
+    cnvi_top: Optional[IntelTLV]
+    cnvr_top: Optional[IntelTLV]
+    cnvi_bt: Optional[IntelTLV]
+    img_type: Optional[IntelTLV]
+    sbe_type: Optional[IntelTLV]
+
+
+def parse_intel_version_tlvs(data: bytearray) -> IntelVersionTlv:
+    tlvs = {
+        'cnvi_top': 0x10,
+        'cnvr_top': 0x11,
+        'cnvi_bt': 0x12,
+        'img_type': 0x1C,
+        'sbe_type': 0x2F,
+    }
+
+    parsed_tlvs: Dict[str, Optional[IntelTLV]] = {key: None for key in tlvs.keys()}
+    i = 0
+    while i < len(data):
+        try:
+            tlv_type = data[i]
+            tlv_length = data[i + 1]
+            tlv_value = data[i + 2 : i + 2 + tlv_length]
+
+            for name, type_code in tlvs.items():
+                if tlv_type == type_code:
+                    parsed_tlvs[name] = IntelTLV(tlv_type, tlv_length, tlv_value)
+                    break
+
+            i += 2 + tlv_length
+        except IndexError:
+            logging.error("TLV parse error")
+            continue
+
+    return IntelVersionTlv(
+        cnvi_top=parsed_tlvs.get('cnvi_top'),
+        cnvr_top=parsed_tlvs.get('cnvr_top'),
+        cnvi_bt=parsed_tlvs.get('cnvi_bt'),
+        img_type=parsed_tlvs.get('img_type'),
+        sbe_type=parsed_tlvs.get('sbe_type'),
+    )
+
+
+async def secure_send(host, fragment_type: int, plen: int, param: bytes):
+    while plen > 0:
+        fragment_len = MAX_FRAGMENT_PAYLOAD if plen > MAX_FRAGMENT_PAYLOAD else plen
+        cmd_param = bytearray(fragment_len + 1)
```

Comment:
suggest: `bytearray` isn't needed here, you can just construct: `cmd_param = bytes([fragment_type]) + param[:fragment_len]`